### PR TITLE
android: egui editor

### PR DESCRIPTION
**Pull request by smailbarkouch**
Bringing the egui-editor to android.

Screenshot:
<details>

</details>

### CODE REVIEW

File: clients/android/egui-editor/src/main/jniLibs/arm64-v8a/libegui_editor.so
Comment (Parth) [inline clients/android/egui-editor/src/main/jniLibs/arm64-v8a/libegui_editor.so:1]: Don't let these sneak in

**Review by Parth — CHANGES_REQUESTED** [review]

**Comment by Parth** [comment]
next time you make progress here can you add a gitignore and squash your branch, 
image

these files are slowing down a number of operations that rely on git

File: libs/editor/SwiftEditor/Sources/Editor/iOSMTK.swift [hunk 1]
```diff
@@ -389,6 +389,9 @@ public class iOSMTK: MTKView, MTKViewDelegate, UITextInput, UIEditMenuInteractio
         let point = Unmanaged.passUnretained(touches.first!).toOpaque()
         let value = UInt64(UInt(bitPattern: point))
         let location = touches.first!.location(in: self)
+        
+        print("press at \(location.x) \(location.y)")
```
Comment (tvanderstad) [inline libs/editor/SwiftEditor/Sources/Editor/iOSMTK.swift:393]: 🤔

File: libs/editor/egui_editor/src/android/java_ffi.rs [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+    let Some(key) = AndroidKeys::from(key_code) else {
+        return
+    };
+
+    if let Some(key) = key.egui_key() {
+        obj.raw_input
+            .events
+            .push(Event::Key { key, pressed: pressed == 1, repeat: false, modifiers });
+    }
```
Comment (tvanderstad) [inline libs/editor/egui_editor/src/android/java_ffi.rs:270]: may have to detect more situations here https://github.com/lockbook/lockbook/blob/65dea5db545af7b583e088c7e79e700e9feea6dc/libs/editor/egui_editor/src/apple/macos_ffi.rs#L22-L41

File: libs/editor/egui_editor/src/android/keyboard.rs [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+            | Key9 | Apostrophe | Backslash | Comma | Equals | LeftBracket | Minus | Period
+            | RightBracket | Semicolon | Space => true,
+
+            _  => false,
+        }
+    }
+
+    pub fn egui_key(&self) -> Option<egui::Key> {
+        use AndroidKeys::*;
+        let key = match self {
```
Comment (tvanderstad) [inline libs/editor/egui_editor/src/android/keyboard.rs:607]: I belive you're missing Backspace and End

File: libs/editor/egui_editor/src/android/keyboard.rs [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+            _ => None,
+        }
+    }
+
+    pub fn valid_text(&self) -> bool {
+        use AndroidKeys::*;
+        match self {
+            A | B | C | D | E | F | G | H | I | J | K | L | M | N | O | P | Q | R | S | T | U
+            | V | W | X | Y | Z | Key0 | Key1 | Key2 | Key3 | Key4 | Key5 | Key6 | Key7 | Key8
+            | Key9 | Apostrophe | Backslash | Comma | Equals | LeftBracket | Minus | Period
```
Comment (tvanderstad) [inline libs/editor/egui_editor/src/android/keyboard.rs:598]: missing ` and /

File: libs/editor/egui_editor/src/android/keyboard.rs (hunk 3 above)
Comment (smailbarkouch) [reply libs/editor/egui_editor/src/android/keyboard.rs:607]: You're right thanks for the catch.

File: libs/editor/egui_editor/src/android/java_ffi.rs (hunk 2 above)
Comment (smailbarkouch) [reply libs/editor/egui_editor/src/android/java_ffi.rs:270]: Added, thanks.

**Review by Parth — APPROVED** [review]

**Pull request merged**